process/block/preprocess/transactions.go

```diff
@@ -66,8 +66,12 @@ type transactions struct {
 	emptyAddress                 []byte
 	txTypeHandler                process.TxTypeHandler
 	scheduledTxsExecutionHandler process.ScheduledTxsExecutionHandler
+	senderInfo                   map[string]*senderInfo
```

Comment:
please move senderInfo and mutSenderInfo into a small component that can expose private functions. This way we won't be tempted to work with this mutex inside another mutex-controlled area that might lead to ugly deadlocks

Reply:
+1

Reply:
Done